align/schema/constraint.py

```diff
@@ -1328,11 +1360,14 @@ class ConstraintDB(types.List[ConstraintType]):
 
     @types.validate_arguments
     def append(self, constraint: ConstraintType):
-        if hasattr(constraint, 'translate'):
-            if self.parent._checker is None:
-                self.parent.verify()
-            self.parent.verify(constraint=constraint)
-        super().append(constraint)
+        if any(repr(constraint) == repr(x) for x in self):
+            logger.warning(f"Constraint is duplicated: {repr(constraint)}")
```

Comment:
@parijatm , throwing a warning for duplicated constraints.

Reply:
Can we do this only for user constraints instead of doing it globally?

I still feel like duplicate constraints indicate something is actually wrong with our constraint annotation flow. 

Checking for repr will also only allow us to detect a very tiny subset of duplicated constraints. For instance:
```Align(instances=['M1', 'M2', 'M3'], direction='H')```
and
```Align(instances=['M1', 'M3', 'M2'], direction='H')```
will generate two different hashes (and hence will pass the repr test here  and not break the Z3 solver). But they are really the same exact constraint. Need to do something smarter here in the long run.

Reply:
I am assuming that you are referring to the constraints provided by user (but not necessarily of type UserConstraint). I agree that compiler should not generate a duplicate constraint. Is it okay if this is addressed in a future PR?

Reply:
> I am assuming that you are referring to the constraints provided by user (but not necessarily of type UserConstraint).

That is correct.

> I agree that compiler should not generate a duplicate constraint. Is it okay if this is addressed in a future PR?

Certainly !